src/doc/markdown/page12_rest_api.py

```diff
@@ -0,0 +1,438 @@
+The URL expected for the full variant of the API is:
+
+```
+https://fleet<N>.jaia.tech/jaia/v1
+```
+
+All data are expected to be sent as a POST request using the JSON formatted version of the APIRequest message as the POST variable 'json'. For example, to request status from all the bots you would POST the following to the above URL, for example using Python to command Fleet0:
+
+
+```
+#!/usr/bin/env python3
+import requests
+
+jaia_request={"target": {"all": True}, "status": True, "api_key": "4vS6s2jnulxVjrKSB-__tQ"}
+
+res = requests.post(f'https://fleet0.jaia.tech/jaia/v1', json=jaia_request)
+```
+
+"""
+    content += f"""
+Key for placeholders: 
+ - `<API_KEY_STRING>`: API Key string field
+ - `<STRING>`: String field
+ - `False`: Boolean field
+ - `{float_dummy}`: Floating point field
+ - `{max_int32}`: Int32 field
+ - `{max_uint32}`: UInt32 field
+ - `{max_int64}`: Int64 field
+ - `{max_uint64}`: UInt64 field
+"""
+    
+
+    content += """
+See the action documentation below for more details.
+"""
+    
+    for field in jaiabot.messages.rest_api_pb2.APIRequest.DESCRIPTOR.oneofs_by_name['action'].fields:
+        content += generate_action(field.name)
+    for field in jaiabot.messages.rest_api_pb2.APIResponse.DESCRIPTOR.oneofs_by_name['action'].fields:
+        content += generate_response_action(field.name)
+    return content
+
+def generate_action(action):
+    field = jaiabot.messages.rest_api_pb2.APIRequest.DESCRIPTOR.fields_by_name[action]
+    doc = field.GetOptions().Extensions[jaiabot.messages.option_extensions_pb2.field].rest_api.doc
+    
+    content = f"""
+## Request Action: {action}
+
+{doc}
+
+"""
+
+    content += generate_simple_variant(action)
+    content += generate_full_variant(action)
+    
+    return content
+
+def generate_response_action(action):
+    field = jaiabot.messages.rest_api_pb2.APIResponse.DESCRIPTOR.fields_by_name[action]
+    doc = field.GetOptions().Extensions[jaiabot.messages.option_extensions_pb2.field].rest_api.doc
+    
+    content = f"""
+## Response Action: {action}
+
+{doc}
+
+"""
+    content += generate_response(action)
+    
+    return content
+
+
+def generate_simple_variant(action):
+    get_vars=dict()
+    get_vars["api_key"]="<API_KEY_STRING>"
+    
+    action_field_desc = jaiabot.messages.rest_api_pb2.APIRequest.DESCRIPTOR.fields_by_name[action]
+    jaia_request = jaiabot.messages.rest_api_pb2.APIRequest()
+
+    enums=dict()
+    
+    if action_field_desc.type != google.protobuf.descriptor.FieldDescriptor.TYPE_BOOL:
+        jaia_request_action = getattr(jaia_request, action_field_desc.name)
+        for field in jaia_request_action.DESCRIPTOR.fields:
+            presence = field.GetOptions().Extensions[jaiabot.messages.option_extensions_pb2.field].rest_api.presence
+            if presence != jaiabot.messages.option_extensions_pb2.RestAPI.GUARANTEED:
+                continue
+            elif field.label == google.protobuf.descriptor.FieldDescriptor.LABEL_REPEATED:
```

Comment:
The field names are coming out in camelCase here in the generated docs, whereas we're using snake_case for our requests.  Otherwise, this looks really great so far.

Reply:
Good catch - fixed.